contracts/token/Minter.sol

```diff
@@ -34,15 +46,25 @@ contract Minter is Manager, IMinter {
     }
 
     /*
-     * @dev Mint new tokens based on the active stake proportional to the total active stake
-     * @param _activeStake Stake of active transcoder
-     * @param _totalActiveStake Total stake of all active transcoders
+     * @dev Create reward based on a fractional portion of the mintable tokens and redistributable funds for a round
+     * @param _fracNum Numerator of fraction
+     * @param _fracDenom Denominator of fraction
      */
-    function mint(uint256 _activeStake, uint256 _totalActiveStake) external onlyBondingManager returns (uint256) {
-        uint256 amount = mintedTokensPerRound().mul(_activeStake).div(_totalActiveStake);
-        livepeerToken().mint(this, amount);
-
-        return amount;
+    function createReward(uint256 _fracNum, uint256 _fracDenom) external onlyBondingManager returns (uint256) {
+        // Compute fraction of redistributable tokens to include in reward
+        uint256 redistributeAmount = currentRedistributableTokens.mul(_fracNum).div(_fracDenom);
+        // Update amount of redistributable tokens for round
+        currentRedistributableTokens = currentRedistributableTokens.sub(redistributeAmount);
+        redistributionPool = redistributionPool.sub(redistributeAmount);
+
+        // Compute and mint fraction of mintable tokens to include in reward
+        uint256 mintAmount = currentMintableTokens.mul(_fracNum).div(_fracDenom);
+        // Update amount of mintable tokens for round
+        currentMintableTokens = currentMintableTokens.sub(mintAmount);
```

Comment:
if `currentMintableTokens` gets reduced here after we've minted some tokens to one caller of reward, then does that screw up calculations for the next caller? I'm guessing the num and denom are the callers total stake and the total overall stake?

For example, if there are 100 mintable tokens, and two people each have 10% of stake, then you'd want them each to get 10 tokens. By this logic would the second caller get 10% of 90 remaining `currentMintableTokens` and only end up with 9 tokens? It's possible I'm following it wrong.

Reply:
Ah good catch...will need to fix this